### ISSUE TITLE
[Android] Blank keyboard on Android 4.4.2

### ISSUE THREAD

**Issue by darcywong00** [keymanapp/keyman#2323]
A user updated to Keyman 12.0.4206 stable but gets a blank SIL Euro Latin keyboard. All other Keyman keyboards also fail to load.

When connected with a test build, Chrome listed console errors
```
Uncaught SyntaxError: Unexpected identifier keyboard.html:270
Uncaught ReferenceError: Map is not defined keyman.js:18986
```

The SyntaxError is the use of `let` (ES6) and I think the browser for 4.4.2 is ES5.

The ReferenceError correponds to lm-layer's use of Map():
```
var PromiseStore = function() {
  this._promises = new Map();
```

Perhaps the webView on that version of Android doesn't support Map()?

I loaded Keyman for Android 11.0.2106 stable and that keyboard loaded fine.

### Configuration
Device: Samsung S4
OS: Android 4.4.2 (SDK 19)
Keyman: 12.0.4206
Keyboards: SIL Euro Latin (user also wanted SIL IPA and Greek)

**Comment by darcywong00** [comment]
From standup, we'll try to have @jahorton look into this next sprint.
Maybe it's just a matter of having lm-layer target es5?

**Comment by jahorton** [comment]
It appears that the transition point here is API 24 (Android 7.0.0) - before that, the old Browser icon is in place and `Map` doesn't work.  At API 24 onward, the default is Chrome.

**Comment by jahorton** [comment]
Keep in mind that the use of `Map` is on the KMW side - it's used by LMLayer code to coordinate fulfillment of `Promise`s across the WebWorker divide.  We may need to add a polyfill here.

**Comment by darcywong00** [comment]
As of the current stable-12.0, there's still a KMW / lm-layer exception being thrown
```
Uncaught TypeError: Object t has no method 'codePointAt'
```

**Comment by jahorton** [comment]
I hadn't quite caught this one before since I was using MTNT 0.1.2 most of the time.  I noticed the error on IE when implementing #2422 / #2405, but didn't think about how Android may have related issues.

Relevant reference:  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/codePointAt#Browser_compatibility

A bit higher up on the page, there's a pretty short and simple polyfill for the missing method; we can probably just add that.

**Comment by mcdurdin** [comment]
@darcywong00 are you able to test this?

**Comment by darcywong00** [comment]
> are you able to test this?

So testing Keyman for Android [12.0.4213](https://downloads.keyman.com/android/stable/12.0.4213/keyman-12.0.4213.apk) on an emulator with Android API 19, the keyboard fails with this KMW exception:
```
Uncaught TypeError: Cannot convert object to primitive value
```
and from this keyboard.html exception:
```
keyboard.html, keyboardName=EuroLatin (SIL) Keyboard, keyboardType=INAPP, cm_message=Uncaught TypeError: Cannot call method 'init' of undefined, packageID=cloud
```
(KMW not loading?)

------

Even more interesting, when I run the debug build of `stable-12.0`, that keyboard loads fine with the model.

**Comment by jahorton** [comment]
Sigh... this one just keeps getting better.

> we'll still have to keep an eye on https://github.com/keymanapp/keyman/issues/2323
The stable-12 branch has the polyfills, but the release builds are still giving a blank keyboard

**Comment by darcywong00** [comment]
Fixed in Keyman for Android 12.0.4215 stable
